src/components/NavBar.js

```diff
@@ -1,14 +1,24 @@
-import React, { Component } from 'react';
-import { NavLink } from 'react-router-dom';
-import { Container, Menu, Input } from 'semantic-ui-react';
-import '../styling/Navbar.css';
-import AlertMessage from './AlertMessage';
-import { connect } from 'react-redux';
+import React, { Component } from "react";
+import { NavLink } from "react-router-dom";
+import { Container, Menu, Input } from "semantic-ui-react";
+import "../styling/Navbar.css";
+import AlertMessage from "./AlertMessage";
+import { connect } from "react-redux";
+import UpdateLanguage from './UpdateLanguage';
+import i18n from "../i18n";
 
 class NavBar extends Component {
   state = { activeItem: 'latest news' }
+  constructor(props) {
+    super(props)
+    this.ReRender = this.ReRender.bind(this);
+  };
+
+  handleItemClick = (e, { name }) => this.setState({ activeItem: name });
 
-  handleItemClick = (e, { name }) => this.setState({ activeItem: name })
+  ReRender() {
```

Comment:
IF you keep this function, please use conventional naming: `ReRender` should be `reRender`